webapi/src/main/scala/org/knora/webapi/messages/admin/responder/usersmessages/UsersMessagesADM.scala

```diff
@@ -0,0 +1,767 @@
+
+/**
+  * Requests removing the user from a group.
+  *
+  * @param userIri        the IRI of the user to be updated.
+  * @param groupIri       the IRI of the group.
+  * @param requestingUser the user initiating the request.
+  * @param apiRequestID   the ID of the API request.
+  */
+case class UserGroupMembershipRemoveRequestADM(userIri: IRI,
+                                               groupIri: IRI,
+                                               requestingUser: UserADM,
+                                               apiRequestID: UUID) extends UsersResponderRequestADM
+
+
+// Responses
+
+/**
+  * Represents an answer to a request for a list of all users.
+  *
+  * @param users a sequence of user profiles of the requested type.
+  */
+case class UsersGetResponseADM(users: Seq[UserADM]) extends KnoraResponseADM {
+    def toJsValue = UsersADMJsonProtocol.usersGetResponseADMFormat.write(this)
+}
+
+/**
+  * Represents an answer to a user profile request.
+  *
+  * @param user the user's information of the requested type.
+  */
+case class UserResponseADM(user: UserADM) extends KnoraResponseADM {
+    def toJsValue: JsValue = UsersADMJsonProtocol.userProfileResponseADMFormat.write(this)
+}
+
+/**
+  * Represents an answer to a request for a list of all projects the user is member of.
+  *
+  * @param projects a sequence of projects the user is member of.
+  */
+case class UserProjectMembershipsGetResponseADM(projects: Seq[ProjectADM]) extends KnoraResponseADM {
+    def toJsValue: JsValue = UsersADMJsonProtocol.userProjectMembershipsGetResponseADMFormat.write(this)
+}
+
+/**
+  * Represents an answer to a request for a list of all projects the user is member of the project admin group.
+  *
+  * @param projects a sequence of projects the user is member of the project admin group.
+  */
+case class UserProjectAdminMembershipsGetResponseADM(projects: Seq[ProjectADM]) extends KnoraResponseADM {
+    def toJsValue: JsValue = UsersADMJsonProtocol.userProjectAdminMembershipsGetResponseADMFormat.write(this)
+}
+
+/**
+  * Represents an answer to a request for a list of all groups the user is member of.
+  *
+  * @param groups a sequence of groups the user is member of.
+  */
+case class UserGroupMembershipsGetResponseADM(groups: Seq[GroupADM]) extends KnoraResponseADM {
+    def toJsValue: JsValue = UsersADMJsonProtocol.userGroupMembershipsGetResponseADMFormat.write(this)
+}
+
+/**
+  * Represents an answer to a user creating/modifying operation.
+  *
+  * @param user the new user profile of the created/modified user.
+  */
+case class UserOperationResponseADM(user: UserADM) extends KnoraResponseADM {
+    def toJsValue: JsValue = UsersADMJsonProtocol.userOperationResponseADMFormat.write(this)
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Components of messages
+
+/**
+  * Represents a user's profile.
```

Comment:
I see that `UserADM` can convert itself to a `UserProfileV1`. That seems good to me for the `v1` code, but currently, `UserProfileV1` is also used throughout the `v2` code. Should the `v2` code be refactored to use `UserADM` directly?

Reply:
This would be preferable since `UserADM` will be the new main format. Also, `UserADM` will be further extended, while `UserProfileV1` will stay as it is now.